core/src/main/scala/org/apache/spark/resource/ResourceProfileBuilder.scala

```diff
@@ -17,71 +17,67 @@
+ * Resource profile builder to build a [[ResourceProfile]] to associate with an RDD.
+ * A [[ResourceProfile]] allows the user to specify executor and task resource requirements
+ * for an RDD that will get applied during a stage. This allows the user to change the resource
  * requirements between stages.
  *
  */
 @Evolving
 @Since("3.1.0")
 class ResourceProfileBuilder() {
 
+  // Task resource requests that specified by users, mapped from resource name to the request.
   private val _taskResources = new ConcurrentHashMap[String, TaskResourceRequest]()
+  // Executor resource requests that specified by users, mapped from resource name to the request.
   private val _executorResources = new ConcurrentHashMap[String, ExecutorResourceRequest]()
 
-  def taskResources: Map[String, TaskResourceRequest] = _taskResources.asScala.toMap
-  def executorResources: Map[String, ExecutorResourceRequest] = _executorResources.asScala.toMap
-
-  /**
-   * (Java-specific) gets a Java Map of resources to TaskResourceRequest
-   */
-  def taskResourcesJMap: JMap[String, TaskResourceRequest] = _taskResources.asScala.asJava
-
   /**
-   * (Java-specific) gets a Java Map of resources to ExecutorResourceRequest
+   * Add executor resource requests
+   * @param requests The detailed executor resource requests, see [[ExecutorResourceRequests]]
+   * @return this.type
    */
-  def executorResourcesJMap: JMap[String, ExecutorResourceRequest] = {
-    _executorResources.asScala.asJava
-  }
-
-  def require(requests: ExecutorResourceRequests): this.type = {
+  def executorRequire(requests: ExecutorResourceRequests): this.type = {
     _executorResources.putAll(requests.requests.asJava)
     this
   }
 
-  def require(requests: TaskResourceRequests): this.type = {
+  /**
+   * Add task resource requests
+   * @param requests The detailed task resource requests, see [[TaskResourceRequest]]
+   * @return this.type
+   */
+  def taskRequire(requests: TaskResourceRequests): this.type = {
     _taskResources.putAll(requests.requests.asJava)
     this
   }
 
-  def clearExecutorResourceRequests(): this.type = {
-    _executorResources.clear()
-    this
-  }
-
-  def clearTaskResourceRequests(): this.type = {
-    _taskResources.clear()
-    this
-  }
-
   override def toString(): String = {
     "Profile executor resources: " +
       s"${_executorResources.asScala.map(pair => s"${pair._1}=${pair._2.toString()}")}, " +
       s"task resources: ${_taskResources.asScala.map(pair => s"${pair._1}=${pair._2.toString()}")}"
   }
 
-  def build: ResourceProfile = {
-    new ResourceProfile(executorResources, taskResources)
+  def build(): ResourceProfile = {
+    if (!Utils.isTesting) {
+      if (_taskResources.isEmpty) {
+        throw new SparkException("Empty task resource request.")
+      }
+      if (_executorResources.isEmpty) {
+        throw new SparkException("Empty executor resource request.")
+      }
```

Comment:
This seems like a new constraint, can you explain why?

Reply:
I think users should at least assign cores to executors, otherwise, executors can fail to launch.

I have reverted this change to make this PR clean according to @tgravescs 's feedback.